src/azure-cli/azure/cli/command_modules/hdinsight/custom.py

```diff
@@ -490,3 +526,201 @@ def execute_hdi_script_action(cmd, client, resource_group_name, cluster_name,
+
+
+def update_autoscale_condition(cmd, client, resource_group_name, cluster_name, index, days=None, time=None, count=None):
+    autoscale_configuration = None
+    cluster = client.get(resource_group_name, cluster_name)
+    for role in cluster.properties.compute_profile.roles:
+        if role.name == "workernode":
+            autoscale_configuration = role.autoscale_configuration
+            break
+    if not autoscale_configuration:
+        raise CLIError('The cluster has not enabled autoscale.')
+
+    if not autoscale_configuration.recurrence:
+        raise CLIError('The cluster does not have any schedule conditions to update.')
+
+    index = int(index)
+    if index >= len(autoscale_configuration.recurrence.schedule):
+        raise CLIError(
+            'This cluster only has {} condition(s). Please specify a correct index which starts with 0.'.format(
+                len(autoscale_configuration.recurrence.schedule)))
+
+    # try to update schedule condition
+    if days:
+        autoscale_configuration.recurrence.schedule[index].days = days
+    if time:
+        autoscale_configuration.recurrence.schedule[index].time_and_capacity.time = time
+    if count:
+        autoscale_configuration.recurrence.schedule[index].time_and_capacity.min_instance_count = count
+        autoscale_configuration.recurrence.schedule[index].time_and_capacity.max_instance_count = count
+
+    return client.update_auto_scale_configuration(resource_group_name, cluster_name, autoscale_configuration)
+
+
+def delete_autoscale_condition(cmd, client, resource_group_name, cluster_name, index):
+    autoscale_configuration = None
+    cluster = client.get(resource_group_name, cluster_name)
+    for role in cluster.properties.compute_profile.roles:
+        if role.name == "workernode":
+            autoscale_configuration = role.autoscale_configuration
+            break
+    if not autoscale_configuration:
+        raise CLIError('The cluster has not enabled autoscale.')
+
+    if not autoscale_configuration.recurrence:
+        raise CLIError('The cluster does not have any schedule conditions to delete.')
+
+    if len(index) >= len(autoscale_configuration.recurrence.schedule):
+        raise CLIError('Deleting all conditions is not allowed.'
+                       'If you want to disable autoscale please use `az hdinsight autoscale delete`')
+
+    for i in index:
+        i = int(i)
+        if i >= len(autoscale_configuration.recurrence.schedule):
+            raise CLIError(
+                'This cluster only has {} conditions. Please specify a correct index which starts with 0.'.format(
+                    len(autoscale_configuration.recurrence.schedule)))
+        autoscale_configuration.recurrence.schedule.pop(i)
+
+    return client.update_auto_scale_configuration(resource_group_name, cluster_name, autoscale_configuration)
+
+
+def list_autoscale_condition(cmd, client, resource_group_name, cluster_name):
```

Comment:
>list_autoscale_condition [](start = 4, length = 24)

this command seems quite duplicated when `az synapse autoscale show`

Reply:
The response's structure is different. `az hdinsight autoscale show` returns the entire autoscale configuration. 
This command returns an array of schedule conditions.

---
In reply to: [466358083](https://github.com/Azure/azure-cli/pull/14692#discussion_r466358083) [](ancestors = 466358083)

Reply:
This command is only functional when schedule based auto scale is enabled. In this case, customers can use `az hdinsight autoscale show --query` to do the same thing, right?

Reply:
Yes you are right. So do you think I should remove this command?

---
In reply to: [466378211](https://github.com/Azure/azure-cli/pull/14692#discussion_r466378211) [](ancestors = 466378211)

Reply:
Personally I prefer to removing that command.

Reply:
I suggest to keep it. Because it is easier than using `--query`. this cmdlet returns an array of conditions which is helpful for customer to get the index when they want to update and delete condition. From this point, how do you like?

---
In reply to: [466796518](https://github.com/Azure/azure-cli/pull/14692#discussion_r466796518) [](ancestors = 466796518)

Reply:
Hi Dongwei, I added extra attribute 'index' to condition just like monitor [https://github.com/Azure/azure-cli/blob/73823575dd087a8d90814d929458a0804a8553c8/src/azure-cli/azure/cli/command_modules/monitor/operations/autoscale_settings.py#L363](https://github.com/Azure/azure-cli/blob/73823575dd087a8d90814d929458a0804a8553c8/src/azure-cli/azure/cli/command_modules/monitor/operations/autoscale_settings.py#L363)

---
In reply to: [466796518](https://github.com/Azure/azure-cli/pull/14692#discussion_r466796518) [](ancestors = 466796518)

Reply:
removed extra attribute 'index' because  customer may is confused if the index may change when customer run `az hdinsight autoscale condition list`.

---
In reply to: [466915949](https://github.com/Azure/azure-cli/pull/14692#discussion_r466915949) [](ancestors = 466915949,466796518)